src/main/java/com/dev/cinema/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,73 @@
+package com.dev.cinema.dao.impl;
+
+import com.dev.cinema.dao.ShoppingCartDao;
+import com.dev.cinema.exception.DataProcessingException;
+import com.dev.cinema.lib.Dao;
+import com.dev.cinema.model.ShoppingCart;
+import com.dev.cinema.model.User;
+import com.dev.cinema.util.HibernateUtil;
+import org.hibernate.Hibernate;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(shoppingCart);
+            transaction.commit();
+            return shoppingCart;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Cannot add Shopping cart "
+                    + shoppingCart + "into DB", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<ShoppingCart> query = session.createQuery("from ShoppingCart "
+                    + "where id = :userId", ShoppingCart.class);
+            query.setParameter("userId", user.getId());
+            ShoppingCart shoppingCart = query.getSingleResult();
+            Hibernate.initialize(shoppingCart.getTickets());
```

Comment:
I think it's best to avoid using `Hibernate.initialize()`. Use JOIN FETCH in your query instead.

Reply:
done